homeassistant/components/nissan_leaf/__init__.py

```diff
@@ -28,35 +27,31 @@
 from homeassistant.helpers.typing import ConfigType
 from homeassistant.util.dt import utcnow
 
-_LOGGER = logging.getLogger(__name__)
-
-DOMAIN = "nissan_leaf"
-DATA_LEAF = "nissan_leaf_data"
-
-DATA_BATTERY = "battery"
-DATA_CHARGING = "charging"
-DATA_PLUGGED_IN = "plugged_in"
-DATA_CLIMATE = "climate"
-DATA_RANGE_AC = "range_ac_on"
-DATA_RANGE_AC_OFF = "range_ac_off"
-
-CONF_INTERVAL = "update_interval"
-CONF_CHARGING_INTERVAL = "update_interval_charging"
-CONF_CLIMATE_INTERVAL = "update_interval_climate"
-CONF_VALID_REGIONS = ["NNA", "NE", "NCI", "NMA", "NML"]
-CONF_FORCE_MILES = "force_miles"
-
-INITIAL_UPDATE = timedelta(seconds=15)
-MIN_UPDATE_INTERVAL = timedelta(minutes=2)
-DEFAULT_INTERVAL = timedelta(hours=1)
-DEFAULT_CHARGING_INTERVAL = timedelta(minutes=15)
-DEFAULT_CLIMATE_INTERVAL = timedelta(minutes=5)
-RESTRICTED_BATTERY = 2
-RESTRICTED_INTERVAL = timedelta(hours=12)
-
-MAX_RESPONSE_ATTEMPTS = 3
-
-PYCARWINGS2_SLEEP = 30
+from .const import (
+    CONF_CHARGING_INTERVAL,
+    CONF_CLIMATE_INTERVAL,
+    CONF_FORCE_MILES,
+    CONF_INTERVAL,
+    CONF_VALID_REGIONS,
+    DATA_BATTERY,
+    DATA_CHARGING,
+    DATA_CLIMATE,
+    DATA_LEAF,
+    DATA_PLUGGED_IN,
+    DATA_RANGE_AC,
+    DATA_RANGE_AC_OFF,
+    DEFAULT_CHARGING_INTERVAL,
+    DEFAULT_CLIMATE_INTERVAL,
+    DEFAULT_INTERVAL,
+    DOMAIN,
+    INITIAL_UPDATE,
+    LOGGER,
```

Comment:
Are you sure you actually want to make a single logger for the whole integration? This actually lessen the configurability of logging since log configurations are hierarchical. 

Example is that before you could do `homeassistant.components.nissan_leaf.sensor: debug` to only set DEBUG logging level for the `sensor` platform. Or `homeassistant.components.nissan_leaf: debug` for the whole integration.

After this change, you can only do `homeassistant.components.nissan_leaf: debug`.

Reply:
Hi @AngellusMortis, I didn't realise that! Thanks!! I have removed the logging change.

Reply:
I was wondering about that, but I saw many integrations using this pattern and didn't see anything mentioned in the docs, perhaps one @MartinHjelmare or @frenc en can comment and update the docs if needed!

Reply:
I mean, it probably does not matter either way. I doubt any would need that graduality. It was just worth pointing out. It is why most Python code always has one per file and just uses `__name__`.

Some packages get fancy by using specific named logger. Like Django uses `django.request` for request logging.

Reply:
Both ways are accepted for integrations, ie either a logger per module or a logger per package.

Reply:
Thanks @MartinHjelmare.  I'll leave it unchanged for now then.